vstools/enums/color.py

```diff
@@ -794,21 +779,25 @@ def from_res(cls, frame: vs.VideoNode | vs.VideoFrame) -> Primaries:
 
         from ..utils import get_var_infos
 
-        fmt, w, h = get_var_infos(frame)
+        fmt, width, height = get_var_infos(frame)
+
+        # Some matrices are expected to have a certain associated set of primaries
+        # regardless of resolution
+        matrix = Matrix.from_video(frame)
+        primaries = Primaries.from_matrix(matrix)
+        if primaries != Matrix.UNKNOWN:
+            return primaries
 
         if fmt.color_family == vs.RGB:
             return Primaries.BT709
 
-        if w <= 1024 and h <= 576:
-            if h == 576:
-                return Primaries.BT470BG
-
-            return Primaries.ST170M
-
-        if w <= 2048 and h <= 1536:
+        if width >= 1280 or height > 576:
             return Primaries.BT709
-
-        return Primaries.BT2020
+        if height == 576:
```

Comment:
Why is this res dependent but Matrix/Transfer isn't?
Even if it's MPV's behaviour, i'd rather not blindly follow it.